### ISSUE TITLE
ZSTD compression codecs out of experimental feature flag

### ISSUE THREAD

**Issue by sarthakaggarwal97** [opensearch-project/OpenSearch#9502]
**Is your feature request related to a problem? Please describe.**
ZSTD has been moved behind a feature flag with this PR #9476 and the issue #9422.
The idea is to take it out from feature flag so the users will be able to use it without explicitly setting the feature flag in JVM properties.

**Describe the solution you'd like**
The issue will enlist the tasks that are essential to move ZStd Compression Codecs out of the experimental flags.
1. Long running tests to verify for any corruption or memory leaks
2. Move ZStd compression codecs to plugins from core

**Comment by dblock** [comment]
https://github.com/opensearch-project/OpenSearch/pull/9476#pullrequestreview-1594244085

**Comment by sarthakaggarwal97** [comment]
@dblock I agree with you. Most of these checkboxes were covered as a part of testing for #7908, let me know if we should create items for any specific checkboxes so that we don't lose them. Thank you!

**Comment by dblock** [comment]
> @dblock I agree with you. Most of these checkboxes were covered as a part of testing for #7908, let me know if we should create items for any specific checkbox so that we don't lose them. Thank you!

Yes, good idea. Up to you if you want to create the remaining ones as standalone features/issues and reference them back to here or rename this issue as something like "punchlist for zstd" :)
